core/server/master/src/test/java/alluxio/master/AlluxioMasterProcessTest.java

```diff
@@ -209,17 +248,33 @@ public void restoreFromBackupLocal() throws Exception {
   }
 
   private void startStopTest(AlluxioMasterProcess master) throws Exception {
+    startStopTest(master, true, true);
+  }
+
+  private void startStopTest(AlluxioMasterProcess master,
+      boolean expectWebServiceStarted, boolean expectMetricsSinkStarted) throws Exception {
+    waitForAllServingReady(master, 5000);
+    assertTrue(expectWebServiceStarted == master.isWebServing());
+    assertTrue(expectMetricsSinkStarted == master.isMetricSinkServing());
+    master.stop();
+    assertFalse(isBound(mRpcPort));
+    assertFalse(isBound(mWebPort));
+  }
+
+  void waitForAllServingReady(AlluxioMasterProcess master, int timeoutMs)
```

Comment:
nit: It's a bit strange that you allow twice of `timeoutMs`, but it's fine in the test. I don't see a quick better way.